FindTown/FindTown/Presentation/HomeScene/Home/HomeViewController.swift

```diff
@@ -6,102 +6,273 @@
+        
+        townTableView.rowHeight = 150
+        townTableView.estimatedRowHeight = 150
     }
     
-    @objc func action4() {
-        viewModel?.pop()
-    }
-
-    @objc func action2() {
-        viewModel?.present()
+    override func bindViewModel() {
+        
+        filterCollectionView.rx.setDelegate(self)
+            .disposed(by: disposeBag)
+        
+        // Input
+        
+        filterButton.rx.tap
+            .throttle(.milliseconds(300), scheduler: MainScheduler.instance)
+            .bind { [weak self] in
+                self?.viewModel?.input.filterButtonTrigger.onNext(())
+            }
+            .disposed(by: disposeBag)
+        
+        filterResetButton.rx.tap
+            .throttle(.milliseconds(300), scheduler: MainScheduler.instance)
+            .bind { [weak self] in
+                self?.FilterResetButtonHidden(true)
+                self?.viewModel?.input.resetButtonTrigger.onNext(())
+            }
+            .disposed(by: disposeBag)
+        
+        // Output
+        
+        viewModel?.output.searchFilterDataSource
+            .observe(on: MainScheduler.instance)
+            .bind(to: filterCollectionView.rx.items) { collectionView, row, item in
+                let cell = collectionView.dequeueReusableCell(
+                    withReuseIdentifier: FillterCollectionViewCell.reuseIdentifier,
+                    for: IndexPath(row: row, section: 0)
+                ) as! FillterCollectionViewCell
+                
+                cell.setupCell(item, row)
+                
+                return cell
+            }
+            .disposed(by: disposeBag)
+        
+        viewModel?.output.searchTownTableDataSource
+            .observe(on: MainScheduler.instance)
+            .bind(to: townTableView.rx.items) { tableView, row, item in
+                let cell = tableView.dequeueReusableCell(
+                    withIdentifier: TownTableViewCell.reuseIdentifier,
+                    for: IndexPath(row: row, section: 0)
+                ) as! TownTableViewCell
+                
+                cell.setupCell(item)
+                cell.selectionStyle = .none
+                
+                return cell
+            }
+            .disposed(by: disposeBag)
+        
+        viewModel?.output.searchTownTableDataSource
+            .observe(on: MainScheduler.instance)
+            .bind { [weak self] in
+                self?.townCountTitle.text = "\($0.count)개 동네"
+            }
+            .disposed(by: disposeBag)
     }
-
-    @objc func action3() {
-        viewModel?.dismiss()
+    
+    func dismissBottomSheet(_ tempModel: TempFilterModel) {
+        
+        FilterResetButtonHidden(false)
+        
+        let tempTrafiice = tempModel.traffic.isEmpty ? "교통" :
+        tempModel.traffic.count == 1 ? tempModel.traffic.first :
+        "\(tempModel.traffic.first ?? "") 외 \(tempModel.traffic.count-1)개"
+        guard let tempTrafiice else { return }
+        
```

Comment:
```suggestion
       var tempTraffic = "교통"
       if let first = tempModel.traffic.first {
            tempTraffic = tempModel.traffic.count == 1 ? first : first + " 외 \(tempModel.traffic.count - 1) 개"
        }
        
```
여기 코드를 이렇게 개선해보는건 어떨까용?!

Reply:
너무 좋습니다..!! 😆